src/NumberGuessingGame.js

```diff
@@ -1,77 +1,55 @@
-import React, { Component } from "react";
-import GuessControl from "./GuessControl";
-import GuessMessage from "./GuessMessage";
-import GameOver from "./GameOver";
+import GuessControl from "./GuessControl"
+import GuessMessage from "./GuessMessage"
+import GameOver from "./GameOver"
+import { useState } from "react"
 
 /**
  *
  * Returns a random integer number from 1-100 inclusive
  */
 function getRandomNumber() {
-  return Math.floor(Math.random() * 100) + 1;
+  return Math.floor(Math.random() * 100) + 1
 }
 
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 5
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+function NumberGuessingGame() {
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0)
+  const [latestGuess, setLatestGuess] = useState(null)
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber())
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
-
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
+  function handleGuess(guess) {
+    setLatestGuess(guess)
+    setNumberOfGuesses(numberOfGuesses + 1)
   }
 
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
+  function handleReset() {
+    setNumberToGuess(getRandomNumber())
+    setNumberOfGuesses(0)
+    setLatestGuess(null)
   }
 
-  render() {
-    const isCorrectGuess = this.state.latestGuess === this.state.numberToGuess;
-
-    const isGameOver =
-      isCorrectGuess || this.state.numberOfGuesses === MAX_ATTEMPTS;
-
-    return (
-      <div>
-        <h2>I'm thinking of a number from 1 to 100.</h2>
-        <h2>
-          Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
-        </h2>
-        <GuessControl onGuess={this.handleGuess} />
-        {isGameOver && (
-          <GameOver hasWon={isCorrectGuess} onReset={this.handleReset} />
-        )}
-        {!isGameOver && (
-          <GuessMessage
-            guess={this.state.latestGuess}
-            numberToGuess={this.state.numberToGuess}
-            numberOfGuesses={this.state.numberOfGuesses}
-          />
-        )}
-      </div>
-    );
-  }
+  const isCorrectGuess = latestGuess === numberToGuess
+
+  const isGameOver = isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS
```

Comment:
Good job removing the render() function!!

Now, let's try and use the useMemo or useEffect hook to prevent isCorrectGuess and isGameOver to be calculated every re-render.